regolith/tools.py

```diff
@@ -1322,3 +1326,50 @@ def is_fully_appointed(person, begin_date, end_date):
             else:
                 print("appointment gap for {} on {}".format(person.get('_id'), str(day)))
     return status
+
+def key_value_pair_filter(collection, arguments, keys=None):
+    """Retrieves a list of all documents from the collection where the fragment
+        appears in any one of the given fields
+
+        Parameters
+        ----------
+        collection: generator
+            The collection containing the documents
+        arguments: list
+            The name of the fields to look for and their accompanying substring
+        keys: list
+            The name of the fields to return from the search
+
+        Returns
+        -------
+        str:
+            A str of all the ids of the collection concatenated together
+
+        Examples
+        --------
+        >>> key_value_pair_filter(people, ['name', 'ab', 'position', 'professor'], ['_id', 'name'])
+
+        This would get all people for which their name contains the string 'ab'
+        and whose position is professor. It would return the name and id of the
+        valid entries
+
+        """
+    if not keys:
+        keys = ['_id']
+    if '_id' not in keys:
+        keys.insert(0, '_id')
+
+    if len(arguments) % 2 != 0:
+        raise RuntimeError("Error: Number of keys and values do not match")
+    elements = collection
+    for i in range(0, len(arguments) - 1, 2):
+        elements = fragment_retrieval(elements, [arguments[i]], arguments[i + 1])
+    output = ""
+    for e in elements:
+        for key in keys:
+            if key == '_id':
```

Comment:
didn't you already insert '_id' into keys above, so this if clause not needed....